### PR TITLE
Chapter 1 solutions except recursion one

### PR DESCRIPTION
**Pull request by nbelluco**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -542,7 +545,10 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid x y z 
+    | x < y = if y < z then y else z
```
Comment (vrom911) [inline src/Chapter1.hs:549]: Here could be the case that `x > z`, and in this case, you would need to return `x` 🙂

File: src/Chapter1.hs [hunk 2]
```diff
@@ -542,7 +545,10 @@ value after "=" where the condition is true.
...
+    | z < x = if y < z then z else y
+    | otherwise = z
```
Comment (vrom911) [inline src/Chapter1.hs:551]: I noticed, that this function never returns `x`. Probably, needs to modification, as you may want to have something like:

```haskell
ghci> mid 2 1 3
2
```

File: src/Chapter1.hs [hunk 3]
```diff
@@ -620,8 +632,10 @@ Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
-
+sumLast2 n = 
+    let tens = div (mod (abs n) 100) 10 
+        units = lastDigit (abs n)
```
Comment (vrom911) [inline src/Chapter1.hs:637]: Nice reusage of the existing function! 💯

File: src/Chapter1.hs [hunk 4]
```diff
@@ -477,8 +477,8 @@ Implement a function that returns the last digit of a given number.
   results. Or you can try to guess the function name, search for it and check
   whether it works for you!
 -}
--- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit n = mod (abs n) 10
```
Comment (vrom911) [inline src/Chapter1.hs:481]: Perfect 👌🏼

**Review by vrom911 — APPROVED** [review]
Congrats with finishing Chapter One 🎉  Very well done 👏🏼

**Comment by vrom911** [comment]
@nbelluco I have noticed that you started working on Chapter 2 in this PR. I would suggest you to open the new PR for each Chapter, so you would be able to Participate in Hacktoberfest with 4 PRs and also get a more modular review of your solutions 🙂 
This solution is approved, so feel free to close the PR  because we can't merge the solutions to our repo :slightly_smiling_face: But don't worry, it still counts towards Hacktoberfest contribution :shirt: :deciduous_tree: Feel free to open PRs with solutions to other chapters!

**Comment by nbelluco** [comment]
> @nbelluco I have noticed that you started working on Chapter 2 in this PR. I would suggest you to open the new PR for each Chapter, so you would be able to Participate in Hacktoberfest with 4 PRs and also get a more modular review of your solutions slightly_smiling_face
> This solution is approved, so feel free to close the PR because we can't merge the solutions to our repo slightly_smiling_face But don't worry, it still counts towards Hacktoberfest contribution shirt deciduous_tree Feel free to open PRs with solutions to other chapters!

I'm sorry, I'm kinda new to the pull requests mechanism as I'm simply used to merge branches. Thanks for your patience Veronika !

**Comment by vrom911** [comment]
No worries! Let me know if you need any help with that! 😊

**Pull request closed without merging**